client/src/App.jsx

```diff
@@ -67,16 +77,28 @@ class Photos extends React.Component {
 
   render () {
 
+    const isHovering = this.state.zoom;
+    let popover;
+    if (isHovering) {
+      popover = <ZoomPopover primaryPhotoUrl={this.state.primaryPhotoUrl} cordinates={this.state.modalCordinates}></ZoomPopover>
+    } else {
+      popover = null;
+    }
+
   return (
     <div>
       <PhotosWrapper>
         <Thumbnails setPrimary={this.setPrimary} primaryPhotoUrl={this.state.primaryPhotoUrl} photos={this.state.productPhotosUrls}/>
         <PrimaryPhotoWrapper>
-          <PrimaryPhoto onMouseMove={(e) => this.setCordinates(e)} style={{maxWidth: "100%", height: "auto"}} src={this.state.primaryPhotoUrl}></PrimaryPhoto>
-          <Modal primaryPhotoUrl={this.state.primaryPhotoUrl} cordinates={this.state.modalCordinates}></Modal>
+          <PrimaryPhoto
+            onMouseEnter={() => this.toggleZoom()}
+            onMouseLeave={() => this.toggleZoom()}
+            onMouseMove={(e) => this.setCordinates(e)}
+            style={{maxWidth: "100%", height: "auto"}} src={this.state.primaryPhotoUrl}>
```

Comment:
Doesn't the code have the styled defined for this element up above? Could this styling go up above in the styled component for PrimaryPhoto?

Reply:
Good catch! Need to do some code clean-up there

Reply:
Actually, for some reason the styled-component style does not work. So the inline style is required. Would be good to know why that is though.